deployments/mumbai/usdc/deploy.ts

```diff
@@ -0,0 +1,192 @@
+import { Deployed, DeploymentManager } from '../../../plugins/deployment_manager';
+import { DeploySpec, deployComet, exp, sameAddress, wait } from '../../../src/deploy';
+
+const clone = {
+  usdcImpl: '0xa2327a938Febf5FEC13baCFb16Ae10EcBc4cbDCF',
+  usdcProxy: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
+  weth: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
+  wbtc: '0x2260fac5e5542a773aa44fbcfedf7c193bc2c599',
+  wmatic: '0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270',
+  dai: '0x6b175474e89094c44da98b954eedeac495271d0f',
+};
+
+const secondsPerDay = 24 * 60 * 60;
+
+export default async function deploy(deploymentManager: DeploymentManager, deploySpec: DeploySpec): Promise<Deployed> {
+  const deployed = await deployContracts(deploymentManager, deploySpec);
+  await mintTokens(deploymentManager);
+  return deployed;
+}
+
+async function deployContracts(deploymentManager: DeploymentManager, deploySpec: DeploySpec): Promise<Deployed> {
+  const trace = deploymentManager.tracer()
+  const ethers = deploymentManager.hre.ethers;
+  const signer = await deploymentManager.getSigner();
+
+  const fxChild = await deploymentManager.contract('fxChild');
```

Comment:
Sorry about this one too (I think I suggested making this change 🤦 ), but it seems circular to require `fxChild` to exist in roots before we call `deploy`. I think we're betting off hardcoding `fxChild` in the migration script and just setting it in roots at the end of `deploy`. WDYT?

Reply:
i updated `deploymentManager.existing` to allow you to specify a network other than "mainnet", and now we pull in `fxChild` like so and then add it to roots:

```
const fxChild = await deploymentManager.existing('fxChild', FX_CHILD, 'mumbai');
```

deleted the hard-coded mumbai roots